lib/std/Progress.zig

```diff
@@ -190,10 +215,14 @@ pub fn start(self: *Progress, name: []const u8, estimated_total_items: usize) *N
         .context = self,
         .parent = null,
         .name = name,
+        .node_tree_depth = 0,
         .unprotected_estimated_total_items = estimated_total_items,
         .unprotected_completed_items = 0,
     };
+    self.end = 0;
+    self.rows_written = 0;
     self.columns_written = 0;
+    self.max_columns = determineTerminalWidth(self) orelse 0;
```

Comment:
Does this mean that by default this will always write 400 columns if unable to determine a limit?

Reply:
The function `bufWriteLineTruncate` handles `self.max_columns` being zero, allowing full use of the buffer, as you assumed. In addition, setting the value to 0 will still yield an output, as expected.

Reply:
I'd also recommend adding detection on MacOS.

Reply:
Good idea. The issue is that I cannot test it myself because I do not have a MacOS install on hand. Perhaps someone else could test what I'd come up with?

Reply:
> or is it as simple as using the Linux code for MacOS, replacing `std.os.linux` with `std.os.darwin`, or does the API differ in that regard?

This gist I found suggests that to be the case https://gist.github.com/jtriley/1108174

Reply:
If you don't want to work on the MacOS check, you could also just change the default to `100` to match the previous behavior.

Reply:
Sorry for the delayed response. I am interested in developing a Darwin variant, but my personal Mac is too old for Zig master, and I am facing trouble working with Darling/KVM. Nonetheless, I've done some testing and just using the Linux code for Darwin does not work, since the syscalls are different. I had a friend of mine test it for me. Although using `std.os.darwin.ioctl` is probably the way to go, I am not able to find code examples and testing is not possible for the aforementioned reasons.

Setting the default to `100` contradicts the changes made, because I increased the limit to be able to store more content, since it is now possible to display multiple lines. The previous behavior was to print regardless of window size, and setting the value to `0` does exactly that.

Reply:
I managed to get a hold of a macOS system and implemented the changes to add support.